packages/react/src/analytics/integrations/GTM/eventsMapper.js

```diff
@@ -5,63 +5,326 @@
+      products: get(properties, 'products', []).map(getProductData),
+      step: properties.step,
+    };
+  },
+  [eventTypes.CHECKOUT_STEP_COMPLETED]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      step: properties.step,
+      option: properties.option,
+    };
+  },
+  [eventTypes.DELIVERY_METHOD_ADDED]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      ...getCheckoutParametersFromEvent(properties),
+    };
+  },
+  [eventTypes.FILTERS_APPLIED]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      filters: properties.filters
+        ? JSON.stringify(properties.filters)
+        : undefined,
+    };
+  },
+  [eventTypes.FILTERS_CLEARED]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      filters: properties.filters
+        ? JSON.stringify(properties.filters)
+        : undefined,
+    };
+  },
+  [eventTypes.INTERACT_CONTENT]: data => {
+    const properties = getEventProperties(data);
+    const formattedProperties = {};
+
+    Object.keys(properties).forEach(key => {
+      const value = properties[key];
+
+      if (isObject(value)) {
+        return;
+      }
+
+      formattedProperties[snakeCase(key)] = value;
+    });
+
+    return formattedProperties;
+  },
+  [eventTypes.LOGIN]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      method: properties.method,
+    };
+  },
+  [eventTypes.LOGOUT]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      product: getProductData(properties),
+    };
+  },
+  [eventTypes.ORDER_COMPLETED]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      ...getCheckoutParametersFromEvent(properties),
+      total: properties.total,
+      tax: properties.tax,
+      shipping: properties.shipping,
+      addressFinder: properties.addressFinder,
+      checkoutStep: properties.step,
+      deliveryType: properties.deliveryType,
+      paymentType: properties.paymentType,
+      packagingType: properties.packagingType,
+      shippingTier: properties.shippingTier,
+      affiliation: properties.affiliation,
+    };
+  },
+  [eventTypes.ORDER_REFUNDED]: data => {
+    const properties = getEventProperties(data);
+
+    return {
+      products: get(properties, 'products', []).map(getProductData),
+      orderId: properties.orderId,
+    };
+  },
   [eventTypes.PRODUCT_LIST_VIEWED]: data => {
     const properties = getEventProperties(data);
 
     return {
       products: get(properties, 'products', []).map(getProductData),
       list: properties.list,
       currency: properties.currency,
+      from: properties.from,
+      error: properties.error,
+      itemListId: properties.listId,
+      itemListName: properties.list,
+      items: getProductItemsFromEvent(properties),
+      filters: properties.filters
+        ? JSON.stringify(properties.filters)
+        : undefined,
```

Comment:
When the `JSON.stringify` is used in an undefined value it will return an undefined, so what do you think about using `JSON.stringify` directly in this line ?
`filters: JSON.stringify(properties.filters)`

Example of behavior:

<img width="413" alt="Screenshot 2024-01-24 at 10 32 01" src="https://github.com/Farfetch/blackout/assets/44556843/7b1b3d06-a56f-46d1-9ef5-62aa803c139f">

Reply:
If properties.filters is `null`, JSON.stringify will return 'null' which we do not want.